examples/streaming_decode_files.rs

```diff
@@ -0,0 +1,210 @@
+  --joiner ./sherpa-onnx-streaming-zipformer-bilingual-zh-en-2023-02-20/joiner-epoch-99-avg-1.int8.onnx \
+  --tokens ./sherpa-onnx-streaming-zipformer-bilingual-zh-en-2023-02-20/tokens.txt \
+  --model-type zipformer \
+  --rule-fsts ./itn_zh_number.fst \
+  --debug 0 \
+  ./itn-zh-number.wav
+*/
+
+use clap::{arg, Parser};
+use sherpa_rs::common_config::FeatureConfig;
+use sherpa_rs::recognizer::online_recognizer::{
+    OnlineCtcFstDecoderConfig, OnlineModelConfig, OnlineParaformerModelConfig, OnlineRecognizer,
+    OnlineRecognizerConfig, OnlineTransducerModelConfig, OnlineZipformer2CtcModelConfig,
+};
+use sherpa_rs::stream::online_stream::OnlineStream;
+
+/// Streaming
+#[derive(Parser, Debug)]
+#[command(version, about, long_about = None)]
+struct Args {
+    /// Please provide one wave file
+    wave_file: String,
+
+    /// Path to the transducer encoder model
+    #[arg(long, default_value = "")]
+    encoder: String,
+
+    /// Path to the transducer decoder model
+    #[arg(long, default_value = "")]
+    decoder: String,
+
+    /// Path to the transducer joiner model
+    #[arg(long, default_value = "")]
+    joiner: String,
+
+    /// Path to the paraformer encoder model
+    #[arg(long, default_value = "")]
+    paraformer_encoder: String,
+
+    /// Path to the paraformer decoder model
+    #[arg(long, default_value = "")]
+    paraformer_decoder: String,
+
+    /// Path to the zipformer2 CTC model
+    #[arg(long, default_value = "")]
+    zipformer2_ctc: String,
+
+    /// Path to the tokens file
+    #[arg(long, default_value = "")]
+    tokens: String,
+
+    /// Number of threads for computing
+    #[arg(long, default_value = "1")]
+    num_threads: i32,
+
+    /// Whether to show debug message
+    #[arg(long, default_value = "0")]
+    debug: i32,
+
+    /// Optional. Used for loading the model in a faster way
+    #[arg(long)]
+    model_type: Option<String>,
+
+    /// Provider to use
+    #[arg(long, default_value = "cpu")]
+    provider: String,
+
+    /// Decoding method. Possible values: greedy_search, modified_beam_search
+    #[arg(long, default_value = "greedy_search")]
+    decoding_method: String,
+
+    /// Used only when --decoding-method is modified_beam_search
+    #[arg(long, default_value = "4")]
+    max_active_paths: i32,
+
+    /// If not empty, path to rule fst for inverse text normalization
+    #[arg(long, default_value = "")]
+    rule_fsts: String,
+
+    /// If not empty, path to rule fst archives for inverse text normalization
+    #[arg(long, default_value = "")]
+    rule_fars: String,
+}
+
+fn main() {
+    // Parse command-line arguments into `Args` struct
+    let args = Args::parse();
+
+    println!("Reading {}", args.wave_file);
+
+    let (samples, sample_rate) = read_wave(&args.wave_file);
+
+    println!("Initializing recognizer (may take several seconds)");
+    let config = OnlineRecognizerConfig {
```

Comment:
Simplify configs with impl default like in other examples

Reply:
rust's default means that there can be a default case, but this configuration obviously doesn't. The best practice should be to provide a series of new methods, such as new_ transducer, new_paraformer, etc. default, there really isn't a particular default standard.

Reply:
I agree but even using new methods for the structus will be too much expressive for irrelevant fields. so maybe it's worth using default and the examples will show how to use it correctly + sherpa-onnx will throw error with info